packages/@react-spectrum/provider/src/Provider.tsx

```diff
@@ -69,7 +69,11 @@ function Provider(props: ProviderProps, ref: DOMRef<HTMLDivElement>) {
     isRequired,
     isReadOnly,
     validationState
-  });
+  };
+  Object.keys(currentProps).forEach(propKey => currentProps[propKey] === undefined && delete currentProps[propKey]);
```

Comment:
I'm not super thrilled about this solution. I tried the aria-urils/mergeProps() which didn't work for this case.

Reply:
hrm.. could use filter instead? then you create a new object instead of calling delete